src/extension.ts

```diff
@@ -110,4 +113,89 @@ export const activate = () => {
         if (!getExtensionSetting('runOnSave') || reason === vscode.TextDocumentSaveReason.AfterDelay) return
         waitUntil(performFixes())
     })
+
+    vscode.workspace.onDidChangeTextDocument(
+        ({ contentChanges, document }) => {
+            if (!getExtensionSetting("insertMissingCommaOnEnter")) {
+                return;
+            }
+
+            if (
+                document.languageId !== "json" &&
+                document.languageId !== "jsonc"
+            ) {
+                return;
+            }
+
+            if (contentChanges.length === 0) {
+                return;
+            }
+
+            const editor = vscode.window.activeTextEditor;
+
+            if (
+                document.uri !== editor?.document.uri ||
+                ["output"].includes(editor.document.uri.scheme)
+            ) {
+                return;
+            }
+            if (
+                vscode.workspace.fs.isWritableFileSystem(
+                    document.uri.scheme
+                ) === false
+            ) {
+                return;
+            }
+
+            editor.edit((edit) => {
+                contentChanges.forEach(async (content) => {
+                    if (
+                        !content.text.startsWith("\n") &&
```

Comment:
I still don't get the idea of using `startsWith` here. We're doing it only after enter keypress, right? So why don't we use `===` then? It also catches multiline pasting, but why do we need it?

Reply:
Also let's move this check to in outer scope e.g.

```ts
if (contentChanges.some(change => /* check whether not enter keypress change */) return
```

I think this is still valid since we want to do this only after simple Enter keypress. If some edits adding newlines and some not, then most probably changes were done by some extension and not user.

Reply:
> I still don't get the idea of using `startsWith` here. We're doing it only after enter keypress, right? So why don't we use `===` then? It also catches multiline pasting, but why do we need it?

After pressing `Enter` the value of `content.text` will be equal to `\n\r` (or `\n`, depending on the EoL type) plus the spaces following it. If we use trim(), it will remove not only spaces, but also the string EoL itself. So the only option I found is to check with `startsWith`.

Reply:
> After pressing Enter the value of content.text will be equal to \n\r (or \n, depending on the EoL type) plus the spaces following it. 

Makes sense, but why did you switch to `.includes()` then? Anyway why don't we use the following code to be strict:

```ts
const firstTextChange = contentChanges[1]!.text
if (contentChanges.some((change) => change.text !== firstTextChange) || !/^\n\r? *$/.test(firstTextChange)) {
    return
}
```

Reply:
> Makes sense, but why did you switch to .includes() then? 

Because in this situation there is no difference between `includes` and `startsWith`, and I didn't want to use the name startsWithEoL for function because it doesn't sound good 😄

>Anyway why don't we use the following code to be strict:

I don't really understand what this code does. And, again, I don't really want to use regex

Reply:
> I don't really understand what this code does. And, again, I don't really want to use regex

Okay, I see, there is a simpler solution without regexs then:

```ts
const firstTextChange = contentChanges[1]!.text
const firstChangeIsEnterOnly = firstTextChange.startsWith('\n') && firstTextChange.trim() === ''
const changesAreTheSame = contentChanges.some((change) => change.text !== firstTextChange)
if (!changesAreTheSame || !firstChangeIsEnterOnly) {
    return
}

```

@AgentRBY What do you think now?

Reply:
This is important as there is no way to determine wether user did Enter keypress or content pasting so we need to ensure we don't run it when user inserted content that has newline somewhere in the middle.

Reply:
> This is important as there is no way to determine wether user did Enter keypress or content pasting so we need to ensure we don't run it when user pasted content that has newline somewhere in the middle.

Can you tell me a case where your current code doesn't work? 
I have long ago added a check to see if the current line is empty

![image](https://user-images.githubusercontent.com/68181944/203377043-39766a8a-05fc-46e1-b969-0c7f01be3148.png)

And it works quite well

![Code_J7Pe1vL7hX](https://user-images.githubusercontent.com/68181944/203377313-038ca189-80e9-4092-863e-616b3256c74a.gif)

Reply:
I believe you didn't get me right. I was talking about content pasting (when you do ctrl+V the pasted content will be contentChanges`, as you probably mentioned in https://github.com/zardoy/vscode-fix-all-json/pull/6#issuecomment-1324049348). Again,
> content pasting so we need to ensure we don't run it when user pasted content that has newline somewhere in the middle

I want to exclude potential trigger from content pasting (https://github.com/zardoy/vscode-fix-all-json/pull/6#issuecomment-1324049348). Also, the code I provided to check content change `firstTextChange.startsWith('\n') && firstTextChange.trim() === ''` is more explicit rather than `.includes('\n')`

Reply:
I added your check to the last commit. Please check it @zardoy

Reply:
Thanks!